tests/test_spend.py

```diff
@@ -308,42 +316,52 @@ def test_coin_selection(lianad, bitcoind):
     dest_addr_2 = bitcoind.rpc.getnewaddress()
     # If feerate is higher than ancestor, we'll need to pay extra.
 
+    def additional_fees(anc_vsize, anc_fee, target_feerate):
+        """The additional fee which must have been computed by lianad."""
+        computed_anc_vsize = int(anc_fee / target_feerate)
+        print(f"c  ", computed_anc_vsize)
+        extra_vsize = anc_vsize - computed_anc_vsize
+        print("e  ", extra_vsize)
+        return extra_vsize * target_feerate
+
     # Try 10 sat/vb:
-    spend_res_2 = lianad.rpc.createspend({dest_addr_2: 10_000}, [], 10)
+    feerate = 10
+    spend_res_2 = lianad.rpc.createspend({dest_addr_2: 10_000}, [], feerate)
     assert "psbt" in spend_res_2
     spend_psbt_2 = PSBT.from_base64(spend_res_2["psbt"])
     # The spend is using the unconfirmed change.
     assert spend_psbt_2.tx.vin[0].prevout.hash == uint256_from_str(
         bytes.fromhex(spend_txid_1)[::-1]
     )
-    assert bitcoind.rpc.getmempoolentry(spend_txid_1)["ancestorsize"] == 161
-    assert bitcoind.rpc.getmempoolentry(spend_txid_1)["fees"]["ancestor"] * COIN == 339
-    # ancestor vsize at feerate 10 sat/vb = ancestor_fee / 10 = 339 / 10 = 33
-    # extra_weight <= (extra vsize * witness factor) = (161 - 33) * 4 = 512
-    # additional fee at 10 sat/vb (2.5 sat/wu) = 512 * 2.5 = 1280
+    anc_vsize = int(bitcoind.rpc.getmempoolentry(spend_txid_1)["ancestorsize"])
+    anc_fees = int(bitcoind.rpc.getmempoolentry(spend_txid_1)["fees"]["ancestor"] * COIN)
+    if not USE_TAPROOT:
+        assert anc_vsize == 161 and anc_fees == 339
```

Comment:
By removing these assertions about exact values, I'm no longer having any failures. I think it's OK to remove them as the assertions below with `additional_fee` will still be tested.

Reply:
Indeed. I ran this test 200 times both on Taproot and Segwit v0 and didn't have a single failure. I'll drop those in a new commit and mark them as fixing #1000.